dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -293,12 +293,18 @@ public void deleteStep(final WorkflowStep step) throws DotDataException {
 			// Checking for Next Step references
 			for(WorkflowStep otherStep : findSteps(findScheme(step.getSchemeId()))){
 
-				for(WorkflowAction action : findActions(otherStep, APILocator.getUserAPI().getSystemUser())){
-
-					if(action.getNextStep().equals(step.getId())) {
-						throw new DotDataException("</br> <b> Step : '" + step.getName() + "' is being referenced by </b> </br></br>" + 
-								" Step : '"+otherStep.getName() + "' ->  Action : '" + action.getName() + "' </br></br>");
-					}
+				/*
+				Verify we are not validating the next step is the step we want to delete.
+				Remember the step can point to itself and that should not be a restriction when deleting.
+				 */
+				if (!otherStep.getId().equals(step.getId())) {
+					for(WorkflowAction action : findActions(otherStep, APILocator.getUserAPI().getSystemUser())){
+
+                        if(action.getNextStep().equals(step.getId())) {

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)